runtime/onert/core/src/exec/Executors.cc

```diff
@@ -255,24 +256,25 @@ void Executors::executeModels(const IODescription &desc)
       auto output_pkg_index = find_output_index(model_index, ir::SubgraphIndex{0}, ir::IOIndex{i});
       if (output_pkg_index != -1)
       {
-        desc_inter.outputs[i] = std::make_unique<OutputDesc>(*desc.outputs[output_pkg_index].get());
-        continue;
-      }
+        auto output_desc = desc.outputs[output_pkg_index].get();
+        pkgs_outputs[output_pkg_index] = std::make_unique<backend::builtin::IOTensor>(info, layout);
+        pkgs_outputs[output_pkg_index]->setUserTensor(
+          reinterpret_cast<uint8_t *>(output_desc->buffer), output_desc->size);
 
-      curr_bufs[i] = std::make_unique<uint8_t[]>(info.total_size());
-      desc_inter.outputs[i] =
-        std::make_unique<OutputDesc>(info, curr_bufs[i].get(), info.total_size(), layout);
+        outputs_inter[i] = pkgs_outputs[output_pkg_index].get();
+      }
+      else
+      {
+        assert(edge_tensor.size() == edge_bufs.size());
+        const auto temp_index = edge_tensor.size();
+        edge_bufs.emplace_back(std::make_unique<uint8_t[]>(info.total_size()));
+        edge_tensor.emplace_back(std::make_unique<backend::builtin::IOTensor>(info, layout));
+        edge_tensor.at(temp_index)->setUserTensor(edge_bufs[temp_index].get(), info.total_size());
+        outputs_inter[i] = edge_tensor[temp_index].get();
```

Comment:
> > Q2. Isn't it waste memory if pkg uses 3 models and 1st model's output is used as 2nd model's input?
> 
> It's not. Currently, executor's inputs/outputs are always `IPortableTensor` and have to be set buffers externally regardless of backend. Only necessary buffers of internal tensors are created.

Then on example, 1st model's output buffer is allocated here, and when runtime executes 3rd model, it doesn't free useless 1st model's output tensor buffer. Then It may increase peak memory.

Reply:
> Then on example, 1st model's output buffer is allocated here, and when runtime executes 3rd model, it doesn't free useless 1st model's output tensor buffer. Then It may increase peak memory.

Oh, you are right. peak memory may increase. It is more necessary to count the number of edges for each `from` and to free memory when the count becomes 0 after executing the executor.